packages/babel-parser/src/parser/statement.js

```diff
@@ -2068,6 +2087,33 @@ export default class StatementParser extends ExpressionParser {
     node.specifiers.push(this.finishNode(specifier, type));
   }
 
+  maybeParseModuleAttributes() {
+    if (!this.eat(tt._with)) {
+      if (this.hasPlugin("moduleAttributes")) return [];
+      return null;
+    }
+    this.expectPlugin("moduleAttributes");
+    const attrs = [];
+    do {
+      // we are trying to parse a node which has the following syntax
+      // with type: "json"
+      // [with -> keyword], [type -> Identifier], [":" -> token for colon], ["json" -> StringLiteral]
+      const node = this.startNode();
+      node.key = this.parseIdentifier();
```

Comment:
```suggestion
      node.key = this.parseIdentifier(true);
```
To align with Property Name, the identifier name should be literal, that is,
```js
import foo from "bar" with for: "test"
```
should be valid.